airbyte-integrations/connectors/source-hubspot/source_hubspot/source.py

```diff
@@ -44,22 +45,60 @@
     Workflows,
 )
 
+SCOPES = [
+    "automation",
+    "content",
+    "crm.lists.read",
+    "crm.objects.companies.read",
+    "crm.objects.contacts.read",
+    "crm.objects.deals.read",
+    "crm.objects.feedback_submissions.read",
+    "crm.objects.owners.read",
+    "crm.schemas.companies.read",
+    "crm.schemas.contacts.read",
+    "crm.schemas.deals.read",
+    "e-commerce",
+    "files",
+    "files.ui_hidden.read",
+    "forms",
+    "forms-uploaded-files",
+    "sales-email-read",
+    "tickets",
+]
+
 
 class SourceHubspot(AbstractSource):
     def check_connection(self, logger: logging.Logger, config: Mapping[str, Any]) -> Tuple[bool, Optional[Any]]:
         """Check connection"""
+        common_params = self.get_common_params(config=config)
+        if common_params.get("authenticator"):
```

Comment:
shouldn't we do the same in `def streams(self, config: Mapping[str, Any]) -> List[Stream]:`? My suggestion is to move this code to `def get_common_params` and probably rename it

Reply:
I understand your point  you want to make this token checking in another command (not only "check" command)

As for me we don't need to do it
we have exectly one place for checking - "check" command

In subsequent commands we just use token as is

Reply:
will it be the same token? I mean `get_access_token()` can return different tokens from one call(check) to another(discover), right?
can the token be edited - granted additional scopes or revoked some?

Reply:
Majority of Oauth2.0 API(s) grant `scopes` when you get refresh_token

Maybe thare are APIs which allows to downgrade scopes on
refresh_token -> access_token exchanged 
but I did not see such API(s)

Reply:
Okay then